clients/src/test/java/org/apache/kafka/common/requests/ProduceRequestTest.java

```diff
@@ -204,7 +228,7 @@ public void testV6AndBelowCannotUseZStdCompression() {
         ProduceRequestData produceData = new ProduceRequestData()
             .setTopicData(new ProduceRequestData.TopicProduceDataCollection(Collections.singletonList(
                 new ProduceRequestData.TopicProduceData()
-                    .setName("test")
+                    .setTopicId(Uuid.fromString("H3Emm3vW7AKKO4NTRPaCWt"))
```

Comment:
RequestContextTest and RequestQuotaTest still uses `setName` on `TopicProduceData`. Should we change them to use `setTopicId`?

Reply:
I changed all requests that doesn't specifiy old versions to use topic id and left comment next to the ones i didn't change